File: lessons/compiled_lang/java/src/j00.arguments.java
// Not public - see a00.output.java for why.
class J00Arguments {
    public static void main(String[] args) {
        // args.length excludes the program name (unlike argv in C/C++) -
        //  the wrapper script's own Makefile-injected "invoked.as" system
        //  property stands in for argv[0] here (see ../Makefile).
        String scriptName = System.getProperty("invoked.as");

        if (args.length != 2) {
            System.err.println();
            System.err.println("You need to enter two numbers:");
            System.err.println();
            System.err.println("   Usage: " + scriptName + " [num1] [num2]");
            System.err.println();
        } else {
            int sum = Integer.parseInt(args[0]) + Integer.parseInt(args[1]);
            System.out.println("The sum of " + args[0] + " and " + args[1] + " is: " + sum + ".");
        }
    }
}

File: lessons/compiled_lang/rust/src/j00.arguments.rs
use std::env;

fn main() {
    let all_args: Vec<String> = env::args().collect();
    let script_name = &all_args[0];
    let args = &all_args[1..];

    if args.len() != 2 {
        eprintln!();
        eprintln!("You need to enter two numbers:");
        eprintln!();
        eprintln!("   Usage: {script_name} [num1] [num2]");
        eprintln!();
    } else {
        let num1: i32 = args[0].parse().unwrap();
        let num2: i32 = args[1].parse().unwrap();
        println!("The sum of {} and {} is: {}.", args[0], args[1], num1 + num2);
    }
}
